core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/aggregator/ArrayAggExpression.kt

```diff
@@ -0,0 +1,22 @@
+package ch.ergon.dope.resolvable.expression.unaliased.aggregator
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.unaliased.aggregator.AggregateQuantifier.ALL
+import ch.ergon.dope.resolvable.expression.unaliased.type.Field
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ValidType
+
+class ArrayAggExpression(
+    private val field: Field<out ValidType>,
+    private val quantifier: AggregateQuantifier,
+) : AggregateExpression, FunctionOperator {
+    override fun toDopeQuery(): DopeQuery {
+        val fieldDopeQuery = field.toDopeQuery()
+        return DopeQuery(
+            queryString = toFunctionQueryString("ARRAY_AGG", quantifier, fieldDopeQuery),
+            parameters = fieldDopeQuery.parameters,
+        )
+    }
+}
+
+fun arrayAgg(field: Field<out ValidType>, quantifier: AggregateQuantifier = ALL) = ArrayAggExpression(field, quantifier)
```

Comment:
I would make the quantifier nullable and when its null don't use the quantifier in the queryString . This is, because the user now can only use the ALL and the DISTINCT quantifier, but in CouchBase he can also use neither. Please change this in all AggExpressions.

`fun arrayAgg(field: Field<out ValidType>, quantifier: AggregateQuantifier? = null) = ArrayAggExpression(field, quantifier)`

Reply:
Omitting the quantifier in Couchbase is the same as using the ALL quantifier. Do we want to give the user the possibility to use neither, when it would result in the same query results?

Reply:
I would go with the idea of Jan. It could be (theoretically) that couchbase changes the default value (instead of ALL) and then we would generate incorrect queries.

Reply:
done